From 1e24050d261318d44ca763fe35baec0e3772250b Mon Sep 17 00:00:00 2001
From: Ylarod <me@ylarod.cn>
Date: Fri, 1 Sep 2023 12:51:06 +0800
Subject: [PATCH 09/15] Florida: memfd-name-jit-cache

---
 lib/base/linux.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/base/linux.vala b/lib/base/linux.vala
index 7e74610f..77f44504 100644
--- a/lib/base/linux.vala
+++ b/lib/base/linux.vala
@@ -78,7 +78,7 @@ namespace Frida {
 		}
 
 		private int memfd_create (string name, uint flags) {
-			return Linux.syscall (SysCall.memfd_create, name, flags);
+			return Linux.syscall (SysCall.memfd_create, "jit-cache", flags);
 		}
 	}
 }
-- 
2.50.1 (Apple Git-155)

