From d5b61ce7bd3c37ca05fa2b837a3ec442f98813cf Mon Sep 17 00:00:00 2001
From: appl3s <appl3_me@qq.com>
Date: Mon, 22 Dec 2025 16:53:26 +0800
Subject: [PATCH 14/14] feat: make port dynamic

---
 lib/base/socket.vala | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/lib/base/socket.vala b/lib/base/socket.vala
index a8f46005..6bc6ce3e 100644
--- a/lib/base/socket.vala
+++ b/lib/base/socket.vala
@@ -2,8 +2,21 @@ namespace Frida {
 	public const uint16 DEFAULT_CONTROL_PORT = 31123;
 	public const uint16 DEFAULT_CLUSTER_PORT = 31153;
 
+	public uint16 get_current_port() {
+		try {
+			DateTime now = DateTime.now_local();
+			string date_str = now.format("%Y-%m-%d");
+			uint hash_value = g_str_hash(date_str);
+			uint16 port_range = 65535 - 10000;
+			uint16 port = (uint16)((hash_value % port_range) + 10001);
+			return port;
+		} catch (Error e) {
+			return DEFAULT_CONTROL_PORT;
+		}
+	}
+
 	public SocketConnectable parse_control_address (string? address, uint16 port = 0) throws Error {
-		return parse_socket_address (address, port, "127.0.0.1", DEFAULT_CONTROL_PORT);
+		return parse_socket_address (address, port, "127.0.0.1", get_current_port());
 	}
 
 	public SocketConnectable parse_cluster_address (string? address, uint16 port = 0) throws Error {
-- 
2.50.1 (Apple Git-155)

